cmd/test/cmd.go

```diff
@@ -0,0 +1,251 @@
+	"github.com/okteto/okteto/pkg/model"
+	"github.com/okteto/okteto/pkg/okteto"
+	oktetoPath "github.com/okteto/okteto/pkg/path"
+	"github.com/okteto/okteto/pkg/remote"
+)
+
+type Options struct {
+	ManifestPath     string
+	ManifestPathFlag string
+	Namespace        string
+	K8sContext       string
+	Variables        []string
+	Build            bool
+	Timeout          time.Duration
+	Wait             bool
+
+	DeployName        string
+	Dependencies      bool
+	DeployWithoutBash bool
+	DeployInRemote    bool
+	DeployServices    []string
+}
+
+func Test(ctx context.Context, ioCtrl *io.Controller, k8sLogger *io.K8sLogger, at deployCMD.AnalyticsTrackerInterface) *cobra.Command {
+	options := &Options{}
+	cmd := &cobra.Command{
+		Use:          "test",
+		Short:        "Run tests",
+		Hidden:       true,
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, servicesToTest []string) error {
+			stop := make(chan os.Signal, 1)
+			signal.Notify(stop, os.Interrupt)
+			exit := make(chan error, 1)
+
+			go func() {
+				exit <- doRun(ctx, servicesToTest, options, ioCtrl, k8sLogger, at)
+			}()
+			select {
+			case <-stop:
+				oktetoLog.Infof("CTRL+C received, starting shutdown sequence")
+				oktetoLog.Spinner("Shutting down...")
+				oktetoLog.StartSpinner()
+				defer oktetoLog.StopSpinner()
+				return oktetoErrors.ErrIntSig
+			case err := <-exit:
+				return err
+			}
+
+		},
+	}
+
+	cmd.Flags().StringVarP(&options.ManifestPath, "file", "f", "", "path to the okteto manifest file")
+	cmd.Flags().StringVarP(&options.Namespace, "namespace", "n", "", "overwrites the namespace where the development environment is deployed")
+	cmd.Flags().StringVarP(&options.K8sContext, "context", "c", "", "context where the development environment is deployed")
+	cmd.Flags().BoolVarP(&options.Build, "build", "", false, "force build of images before running the tests")
+	cmd.Flags().StringArrayVarP(&options.Variables, "var", "v", []string{}, "set a variable (can be set more than once)")
+	cmd.Flags().DurationVarP(&options.Timeout, "timeout", "t", getDefaultTimeout(), "the length of time to wait for completion, zero means never. Any other values should contain a corresponding time unit e.g. 1s, 2m, 3h ")
+	cmd.Flags().BoolVarP(&options.Wait, "wait", "w", false, "wait until the development environment is deployed (defaults to false)")
+	cmd.Flags().BoolVarP(&options.Dependencies, "dependencies", "", false, "deploy the dependencies from manifest")
+
+	cmd.Flags().StringVar(&options.DeployName, "deploy-name", "", "name of the development environment name to be deployed")
+	cmd.Flags().StringArrayVar(&options.DeployServices, "deploy-services", []string{}, "name of the services of the development environment that should be deployed")
+	cmd.Flags().BoolVarP(&options.DeployWithoutBash, "deploy-no-bash", "", false, "execute commands without bash")
+	cmd.Flags().BoolVarP(&options.DeployInRemote, "deploy-remote", "", false, "force run deploy commands in remote")
```

Comment:
In general I agree with the comments about the deploy flags for this command. My thinking was using this command command for deploys and apps that are already dependent on this and are already deployed using these flags and would fail otherwise. The more we can remove, the better